cartographer_rviz/cartographer_rviz/drawable_submap.h

```diff
@@ -62,19 +65,25 @@ class DrawableSubmap : public QObject {
   bool MaybeFetchTexture(ros::ServiceClient* client);
 
   // Returns whether an RPC is in progress.
-  bool QueryInProgress();
+  bool QueryOrRenderInProgress();
 
   // Sets the alpha of the submap taking into account its slice height and the
   // 'current_tracking_z'.
   void SetAlpha(double current_tracking_z);
+  const int& GetSubmapIndex();
```

Comment:
Simply return the value here (the reference is confusing and doesn't help) and the method can be const. In the case of a cheap accessor like this, we inline the implementation and use variable-like names:

int submap_index() const { return submap_index_; }